core/src/main/java/org/elasticsearch/index/engine/SegmentsStats.java

```diff
@@ -42,6 +47,27 @@
     private long indexWriterMaxMemoryInBytes;
     private long versionMapMemoryInBytes;
     private long bitsetMemoryInBytes;
+    private ImmutableOpenMap<String, Long> fileExtDiskUsage = ImmutableOpenMap.of();
+
+    private static ImmutableOpenMap<String, String> extDescriptions = ImmutableOpenMap.<String, String>builder()
+            .fPut("si", "Segment Info")
+            .fPut("fnm", "Fields")
+            .fPut("fdx", "Field Index")
+            .fPut("fdt", "Field Data")
+            .fPut("tim", "Term Dictionary")
+            .fPut("tip", "Term Index")
+            .fPut("doc", "Frequencies")
+            .fPut("pos", "Positions")
+            .fPut("pay", "Payloads")
+            .fPut("nvd", "Norms")
+            .fPut("nvm", "Norms")
+            .fPut("dvd", "DocValues")
+            .fPut("dvm", "DocValues")
+            .fPut("tvx", "Term Vector Index")
+            .fPut("tvd", "Term Vector Documents")
+            .fPut("tvf", "Term Vector Fields")
+            .fPut("liv", "Live Documents")
+            .build();
```

Comment:
Can you add a comment that this is just "best effort", since Lucene changes its extensions over time?  And that it's relatively harmless if the map is missing an extension, since it just means we don't include the description in the stats we return?